### ISSUE TITLE
Passing allow_val_change=True to wandb.init() prevents sweeps from changing hyperparameters

### ISSUE THREAD

**Issue by tom-andersson** [wandb/client#1159]
I'm not sure if this is strictly a bug, but it's certainly not what I would expect.

* Weights and Biases version: 0.9.3
* Python version: 3.7.6
* Operating System: Linux

### Description

Passing `allow_val_change=True` to `wandb.init()` clamps the `config` values to their defaults when running a sweep, and thus no new values are used. The command line inputs introduced by `wandb` seem to be ignored.

### What I Did

I did the usual of defining `sweep.yaml`, running `wandb sweep sweep.yaml` and `wand agent <agent_id>`. In my `train.py` script, I print `wandb.config` after the `wandb.init()` call, which confirms that the hyperparameters in the sweep are always fixed to the defualt `config` values that I pass to `wandb.init()`. Setting `allow_val_change=False` prevents this unexpected behaviour.

I'd expect `allow_val_change` to simply allow me to overwrite `wandb.config` values in my training script, but still allow sweeps to run and modulate the hyperparameters.

P.S. Thank you very much for making this fantastic package; I'm finally emerging from my cave where I manually tune hyperparameters like a maniac.

**Comment by tyomhak** [comment]
Hey Tom, thank you very much for reporting!
Filing a ticket for this.

**Comment by cvphelps** [comment]
HI there Tom, it sounds like you're passing config values to wandb.init()— could you share your script so we can reproduce this issue?

**Comment by tom-andersson** [comment]
Thanks for the responses guys and sorry for the delay.

It wouldn't be useful for me to share my whole script due to custom code, but I can share the relevant opening lines below (imports included in case someone can spot a potential package issue). 

I'm now also finding that attempting to overwrite default hyperparam values using command line entries has no effect. Is this expected behaviour? I thought `wandb` sweeps modify the default values by calling the script with command line entries? When I run the script with `--seed=50`, say, the result of printing `wandb.config` shows that only the default value of `seed=42` is used.

```
import sys
import os
import numpy as np
import json
import pickle
import argparse
import plotly.express as px
import warnings
import scipy
from sklearn.metrics import accuracy_score
import tensorflow as tf
import time
import regex as re
import xarray as xr
from datetime import datetime
from tensorflow.keras import backend as K
from dateutil.relativedelta import relativedelta
import iris
import cartopy.crs as ccrs
import matplotlib.pyplot as plt
import tensorflow
from tensorflow.keras.callbacks import EarlyStopping
from tensorflow.keras.callbacks import ModelCheckpoint
from tensorflow.keras.models import load_model
import matplotlib as mpl
import wandb
from wandb.keras import WandbCallback


defaults = dict(seed=42, learning_rate=0.0001, filter_size=3,
                n_filters_factor=1, weight_decay=0.,
                batch_size=4, dropout_rate=0.35)

wandb.init(project="my_project_name", allow_val_changes=True, config=defaults)

print('\n\nHyperparams:')
print(wandb.config)
print('\n\n')
```

**Comment by ariG23498** [comment]
Hey @tom-andersson 
I tried reproducing the code. I found a simple synatx error in your code snippet otherwise the code seems to be doing what is expected.
## Train.py
```python
import wandb
defaults = dict(seed=42, filter_size=3)
wandb.init(entity="xxx", allow_val_change=True, project="xxx",config=defaults)
print('\n\nHyperparams:')
print(wandb.config)
print('\n\n')
log = 2*wandb.config["filter_size"] - 0.01*wandb.config["seed"]
wandb.log({"output":log})
wandb.finish()
```
## Sweep.yml
```yaml
project: xxx
entity: xxx
program: train.py
method: bayes
metric:
  goal: minimize
  name: output
parameters:
  seed:
    distribution: int_uniform
    max: 100
    min: 42
  filter_size:
    distribution: int_uniform
    max: 7
    min: 3
```
## Code error
```python
wandb.init(project="my_project_name", allow_val_changes=True, config=defaults)
```
The parameter should have been `allow_val_change` instead of `allow_val_changes`.

When I use the `allow_val_change=True` the print statements seem to be printing out the different paramter values as should be
![image](https://user-images.githubusercontent.com/36856589/101144818-e737d400-363e-11eb-8cd4-f157146d81ab.png)
![image](https://user-images.githubusercontent.com/36856589/101144890-00d91b80-363f-11eb-93c6-a1285459df54.png)
![image](https://user-images.githubusercontent.com/36856589/101144954-177f7280-363f-11eb-8134-59d736971090.png)


We are closing this ticket as the solution is found. Please let us know if the problem still persists. :smile:
